docs/core/howto/logger.rst

```diff
@@ -101,37 +101,50 @@ For example:
 This example will show the string "object with value 7 doing a task" because the ``log_source`` key is automatically set to the ``MyObject`` instance that the ``Logger`` is retrieved from.
 
 
-Capturing Failures
-~~~~~~~~~~~~~~~~~~
+Handling Failures
+~~~~~~~~~~~~~~~~~
 
-:py:class:`Logger <twisted.logger.Logger>` provides a :py:meth:`failure <twisted.logger.Logger.failure>` method, which allows one to capture a :py:class:`Failure <twisted.python.failure.Failure>` object conveniently:
+:py:class:`Logger <twisted.logger.Logger>` provides a :py:meth:`failure <twisted.logger.Logger.handlingFailures>` method, which allows one to run some application code, then capture a :py:class:`Failure <twisted.python.failure.Failure>` in the log stream if that code raises an exception:
 
 .. code-block:: python
 
     from twisted.logger import Logger
     log = Logger()
 
-    try:
+    with log.handlingFailures("While doing some math:"):
```

Comment:
> in the log stream if that code raises an exception:

My big comement / issue with this code : `an exception`.

Which exception types ?

Maybe have the API like this ... and it can default to BaseException

```suggestion
    with log.handlingFailures("While doing some math:", (TypeValue, IndexError)):
```

Reply:
I specifically left this out because I don't want this to be used for generalized exception handling. Everywhere that this should be used should be catching all exceptions, and if they're not, then manually catching the exception is not that big a deal.